BookNest/app.js

```diff
@@ -0,0 +1,388 @@
+
+//     // Fetch book details from the database
+//     db.get("SELECT * FROM books WHERE id = ?", [bookId], (err, row) => {
+//         if (err) {
+//             console.error('Error fetching book:', err.message);
+//             return res.status(500).send('Internal Server Error');
+//         }
+
+//         if (!row) {
+//             return res.status(404).send('Book not found');
+//         }
+
+//         // Add the book to the cart
+//         req.session.cart.push(row);
+
+//         // Set a flash message
+//         req.flash('info', 'Book added to cart successfully!');
+//         res.redirect('/'); // Redirect back to the home page
+//     });
+// });
+app.post('/add-to-cart', (req, res) => {
+    const bookId = req.body.bookId;
+    // Fetch book details from the database and add to cart
+    db.get("SELECT * FROM books WHERE id = ?", [bookId], (err, book) => {
+        if (err) {
+            console.error('Database error:', err);
+            return res.status(500).send('Internal Server Error');
+        }
+        if (book) {
+            req.session.cart.push(book); // Add the book to the cart
+            req.session.flashMessage = 'Book added to cart!';
+        }
+        res.redirect('/');
+    });
+});
+
+
+app.get('/login', (req, res) => { // login page: serve index.ejs with login.ejs embedded
+    res.render('index', {
+        title: 'Login',
+        content: 'pages/login'
+    })
+});
+app.post('/login', (req, res) => {
+    const { email, password } = req.body;
+    
+    if (!email || !password) {
+        req.session.flashMessage = 'Email and password are required!';
+        return res.redirect('/login');
+    }
+
+    // Example query to check user
+    const sql = 'SELECT * FROM users WHERE email = ?';
+    db.get(sql, [email], (err, row) => { // Use email instead of username
+        if (err) {
+            console.error('Database error:', err);
+            return res.status(500).send('Database error');
+        }
+        if (!row || row.password !== password) { // Compare with password directly
+            // req.session.flashMessage = 'Invalid email or password';
+            return res.redirect('/login');
+        }
+
+        req.session.user = { id: row.id, email: row.email };
+        res.redirect('/');
+    });
+});
+
+
+app.get('/register', (req, res) => { // register page: serve index.ejs with register.ejs embedded
+    res.render('index', {
+        title: 'Login',
+        content: 'pages/register'
+    })
+});
+// Route: POST /register - Handle user registration
+app.post('/register', (req, res) => {
+    const { username,email, password } = req.body;
+
+    // Log incoming request
+    
+
+    // Validate input: Make sure email and password are not empty
+    if (!email || !password) {
+        req.session.flashMessage = 'Email and password are required.';
+        console.log('Validation failed: Empty email or password.');
+        return res.redirect('/register');
```

Comment:
Register should eventually have clientside & serverside validation to check if: email is valid, user not already registered, and password is secure.